lib/matplotlib/axes/_axes.py

```diff
@@ -6292,8 +6292,8 @@ def _normalize_input(inp, ename='input'):
                         facecolor=c,
                         margins=margins))
             else:
-                for x, y, c in reversed(list(zip(xvals, yvals, color))):
-                    split = 2 * len(bins)
+                split = 2 * len(bins)
+                for x, y, c in reversed(list(zip(xvals, yvals, reversed(color)))):
```

Comment:
That's because py3 went wild over iterators, so `dict.keys()` and similar methods all return iterators.  There is no `__getitem__`, so no way to reverse the order.